### PR TITLE
[poppler] explicitly control dependencies

### PR DESCRIPTION
**Pull request by Thomas1664**
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/poppler/vcpkg.json [hunk 1]
```diff
@@ -83,6 +83,12 @@
         }
       ]
     },
+    "lcms2": {
```
Comment (dg0yt) [inline ports/poppler/vcpkg.json:86]: IMO this feature should be named `cms`. The behaviour is "Use color management system."
The `lcms2` dependency is an implementation detail.
(I know that most other features are named like the related dependencies.)

**Review by dg0yt — COMMENTED** [review]
I wonder if we should patch the `REQUIRED` aspect into `cmake/modules/MacroOptionalFindPackage.cmake`. At the moment, `WITH_FOO` is little more than an inverted `CMAKE_DISABLE_FIND_PACKAGE_FOO`.

**Comment by Thomas1664** [comment]
> I wonder if we should patch the `REQUIRED` aspect into `cmake/modules/MacroOptionalFindPackage.cmake`. At the moment, `WITH_FOO` is little more than an inverted `CMAKE_DISABLE_FIND_PACKAGE_FOO`.

This clearly violates [Prefer options over patching](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#prefer-options-over-patching) because it is possible to do the same thing without patching.

**Comment by dg0yt** [comment]
> > I wonder if we should patch the `REQUIRED` aspect into `cmake/modules/MacroOptionalFindPackage.cmake`. At the moment, `WITH_FOO` is little more than an inverted `CMAKE_DISABLE_FIND_PACKAGE_FOO`.
> 
> This clearly violates [Prefer options over patching](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#prefer-options-over-patching) because it is possible to do the same thing without patching.

This answer was taken account when commenting.
I doubt that the guideline really means to add 20 CMAKE_BLABLA lines (including the necessary `MAYBE_UNUSED_VARIABLES`) instead of changing *one*  line in an 12 years old source files, with potential for upstream acceptance.

(The parameters way would be much easier with #27950.)

**Comment by Thomas1664** [comment]
> This answer was taken account when commenting. I doubt that the guideline really means to add 20 CMAKE_BLABLA lines (including the necessary `MAYBE_UNUSED_VARIABLES`) instead of changing _one_ line in an 12 years old source files, with potential for upstream acceptance.
> 
> (The parameters way would be much easier with #27950.)

I still wonder why you stand so hard behind picky suggestions like this.

Please note that even if I would make the proposed changes there will be a few "CMAKE_BLABLA lines" left. The lines of code you'd save is neglectable. Furthermore, the guidelines are not specific for a certain use case. In general, we should try to avoid patches whenever we can. Furthermore, I don't see what's wrong with ~10 additional lines of CMake options. I'd like to remind you that your PR that improves build time for gettext also added a lot of make-blabla that is technically unnecessary and only achieves a neglectable performance improvement, so I don't understand why you complain so hard here.

Although the macro itself doesn't change, the dependencies that are obtained using it might do. There might be additional dependencies that vcpkg doesn't have. What happens if upstream decides to stop using the macro for a particular dependency? I don't want to dig into possible errors that might happen after this change because I don't have time for that.

In general, I'm against patching at a position that is a central part of a rather complex CMake code and might have unexpected or unintended impacts in the future where people don't pay attention which dependency uses `optional_find_package` and which does not.

**Comment by dg0yt** [comment]
> I still wonder why you stand so hard behind picky suggestions like this.

I would like to stress that my initial wording was carefully to chosen to mark a subjective position, not a hard request. Didn't help, so I'm leaving this discussion.

**Comment by MonicaLiu0311** [comment]
Under tools, only release-generated things are generally placed.(E

image

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by vicroms** [comment]
Let's try to be less confrontational here. Comments and reviews are useful to encourage discussion about implementation and provide additional perspectives to consider. 

Personally, unless they are marked with "Request Changes", I interpret most review comments to be suggestions or subjective opinions.

**Pull request merged**